src/gennylib/src/PoolFactory.cpp

```diff
@@ -192,28 +193,47 @@ std::string PoolFactory::makeUri() const {
 }
 
 mongocxx::options::pool PoolFactory::makeOptions() const {
-    mongocxx::options::tls tlsOptions;
+    mongocxx::options::client clientOptions;
+    auto useTls = _config->getFlag(OptionType::kQueryOption, "tls");
+
+    if (useTls) {
+        mongocxx::options::tls tlsOptions;
+
+        auto allowInv = _config->getFlag(OptionType::kAccessOption, "AllowInvalidCertificates");
+        if (allowInv) {
+            BOOST_LOG_TRIVIAL(debug) << "Allowing invalid certificates for TLS";
+            tlsOptions = tlsOptions.allow_invalid_certificates(true);
+        }
+
+        // Just doing CAFile and PEMKeyFile for now, it's reasonably trivial to add other options
+        // Note that this is entering as a BSON string view, so you cannot delete the config object
+        auto caFile = *_config->get(OptionType::kAccessOption, "CAFile");
+        if (!caFile.empty()) {
+            BOOST_LOG_TRIVIAL(debug) << "Using CA file '" << caFile << "' for TLS";
+            tlsOptions = tlsOptions.ca_file(caFile.data());
+        }
 
-    auto allowInv = _config->getFlag(OptionType::kAccessOption, "AllowInvalidCertificates");
-    if (allowInv) {
-        BOOST_LOG_TRIVIAL(debug) << "Allowing invalid certificates for TLS";
-        tlsOptions = tlsOptions.allow_invalid_certificates(true);
+        auto pemKeyFile = *_config->get(OptionType::kAccessOption, "PEMKeyFile");
+        if (!pemKeyFile.empty()) {
+            BOOST_LOG_TRIVIAL(debug) << "Using PEM Key file '" << pemKeyFile << "' for TLS";
+            tlsOptions = tlsOptions.pem_file(pemKeyFile.data());
+        }
+        BOOST_LOG_TRIVIAL(debug) << "Adding tls options to pool...";
+        clientOptions.tls_opts(tlsOptions);
     }
 
-    // Just doing CAFile and PEMKeyFile for now, it's reasonably trivial to add other options
-    // Note that this is entering as a BSON string view, so you cannot delete the config object
-    auto caFile = *_config->get(OptionType::kAccessOption, "CAFile");
-    if (!caFile.empty()) {
-        BOOST_LOG_TRIVIAL(debug) << "Using CA file '" << caFile << "' for TLS";
-        tlsOptions = tlsOptions.ca_file(caFile.data());
+    if (_config->encryptionCtxt) {
+        BOOST_LOG_TRIVIAL(debug) << "Adding encryption options to pool...";
+        clientOptions.auto_encryption_opts(_config->encryptionCtxt->getAutoEncryptionOptions());
```

Comment:
(nit) If this is the only use of this object here, would it make sense to pass in the encryption options "raw" to this class? Optional up to you.